src/Service/Services/OpenAPI/OpenApiDocumentor.cs

```diff
@@ -0,0 +1,708 @@
+        /// and reference to request body schema.
+        /// </summary>
+        /// <param name="schemaReferenceId">Request body schema object name: For POST requests on entity 
+        /// where the primary key is autogenerated, do not allow PK in post body.</param>
+        /// <param name="requestBodyRequired">True/False, conditioned on whether the table's columns are either
+        /// autogenerated, nullable, or hasDefaultValue</param>
+        /// <returns>Request payload.</returns>
+        private static OpenApiRequestBody CreateOpenApiRequestBodyPayload(string schemaReferenceId, bool requestBodyRequired)
+        {
+            OpenApiRequestBody requestBody = new()
+            {
+                Content = new Dictionary<string, OpenApiMediaType>()
+                {
+                    {
+                        MediaTypeNames.Application.Json,
+                        new()
+                        {
+                            Schema = new OpenApiSchema()
+                            {
+                                Reference = new OpenApiReference()
+                                {
+                                    Type = ReferenceType.Schema,
+                                    Id = schemaReferenceId
+                                }
+                            }
+                        }
+                    }
+                },
+                Required = requestBodyRequired
+            };
+
+            return requestBody;
+        }
+
+        /// <summary>
+        /// Parameters are the placeholders for pk values in curly braces { } in the URL route
+        /// https://localhost:5000/api/Entity/pk1/{pk1}/pk2/{pk2}
+        /// This function creates the string value "/Entity/pk1/{pk1}/pk2/{pk2}"
+        /// and creates associated parameters.
+        /// </summary>
+        /// <param name="entityName">Name of the entity.</param>
+        /// <returns>Primary Key path component. Empty string if no primary keys exist on database object source definition.</returns>
+        private Tuple<string, List<OpenApiParameter>> CreatePrimaryKeyPathComponentAndParameters(string entityName)
+        {
+            SourceDefinition sourceDefinition = _metadataProvider.GetSourceDefinition(entityName);
+            List<OpenApiParameter> parameters = new();
+            StringBuilder pkComponents = new();
+
+            // Each primary key must be represented in the path component.
+            foreach (string column in sourceDefinition.PrimaryKey)
+            {
+                string columnNameForComponent = column;
+
+                if (_metadataProvider.TryGetExposedColumnName(entityName, column, out string? mappedColumnAlias) && !string.IsNullOrEmpty(mappedColumnAlias))
+                {
+                    columnNameForComponent = mappedColumnAlias;
+                }
+
+                if (sourceDefinition.Columns.TryGetValue(columnNameForComponent, out ColumnDefinition? columnDef))
```

Comment:
Do we store the column definition with the key = mapped column name or original column name?

Reply:
wow, yup you're right. Fixed.